cody-shayne/scripts/articleView.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+// Configure a view object, to hold all our functions for dynamic updates and article-related event handlers.
+var articleView = {};
+
+articleView.populateFilters = function() {
+  $('article').each(function() {
+    var authorName, category, optionTag;
+    if (!$(this).hasClass('template')) {
+      // REVIEW: We need to take every author name from the page, and make it an option in the Author filter.
+      //       To do so, Build an `option` DOM element that we can append to the author select box.
+      //       Start by grabbing the author's name from an attribute in `this` article element,
+      //       and then use that bit of text to create the option tag (in a variable named `optionTag`),
+      //       that we can append to the #author-filter select element.
+      authorName = $(this).attr('data-author');
+      optionTag = '<option value="' + authorName + '">' + authorName + '</option>';
+
+      if ($('#author-filter option[value="' + authorName + '"]').length === 0) {
+        $('#author-filter').append(optionTag);
+      }
+
+      // REVIEW: Similar to the above, but...
+      //       Avoid duplicates! We don't want to append the category name if the select
+      //       already has this category as an option!
+      category = $(this).attr('data-category');
+      optionTag = '<option value="' + category + '">' + category + '</option>';
+      if ($('#category-filter option[value="' + category + '"]').length === 0) {
+        $('#category-filter').append(optionTag);
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = function() {
+  $('#author-filter').on('change', function() {
+    // REVIEW: Inside this function, "this" is the element that triggered the event handler function we're
+    //         defining. "$(this)" is using jQuery to select that element, so we can chain jQuery methods
+    //         onto it.
+    if ($(this).val()) {
+      // DONE: If the select box was changed to an option that has a value, we need to hide all the articles,
+      //       and then show just the ones that match for the author that was selected.
+      //       Use an "attribute selector" to find those articles, and fade them in for the reader.
+      $('article').css('display','none');
+      $('article[data-author="' + this.value + '"]').show();
+    } else {
+      $('article').show();
+      $('.template').hide();
+      // DONE: If the select box was changed to an option that is blank, we should
+      //       show all the articles, except the one article we are using as a template.
+
+    }
+    $('#category-filter').val('');
```

Comment:
this is resetting the category filter when an author is selected, you need to do the same thing in the handleCategoryFilter